### ISSUE TITLE
did you using gradient penality in training?

### ISSUE THREAD

**Issue by Johnson-yue** [pomelyu/FUNIT#2]

**Comment by pomelyu** [comment]
Yes, I use gradient penalty on real data, but still could not get promising results. I wonder larger batch size may be helpful.

**Comment by Johnson-yue** [comment]
Oh,  I can not see any code implement **gradient penalty**  , and I think large batch size not helpful because It not contain **BN** in model .

**Comment by pomelyu** [comment]
I had implemented that in the last commit https://github.com/pomelyu/FUNIT/commit/1b5fca18b967f6304866e470e93a75a9700231fa.

**Comment by Johnson-yue** [comment]
@pomelyu  Sorry ,I have no idea  about your this gradient penalty, It is not like WGAN-GP or R1-GP in FUNIT paper .

In FUNIT paper, the real gradient penalty is :
![image](https://user-images.githubusercontent.com/10268274/60636927-011e2780-9e4b-11e9-9b7b-9778fdbfbb66.png)

![image](https://user-images.githubusercontent.com/10268274/60636848-b43a5100-9e4a-11e9-9b23-da5a697cd424.png)

So, What are you according to in your implement?

**Comment by pomelyu** [comment]
I exactly implemented the one described in the paper, but I'm not sure it is correct or not. In Mescheder's paper
![image](https://user-images.githubusercontent.com/2679561/60667615-8a117f00-9e9c-11e9-927c-98889d226b18.png)

And, there is the original implementation https://github.com/LMescheder/GAN_stability/blob/master/gan_training/train.py
```python
def compute_grad2(d_out, x_in):
    batch_size = x_in.size(0)
    grad_dout = autograd.grad(
        outputs=d_out.sum(), inputs=x_in,
        create_graph=True, retain_graph=True, only_inputs=True
    )[0]
    grad_dout2 = grad_dout.pow(2)
    assert(grad_dout2.size() == x_in.size())
    reg = grad_dout2.view(batch_size, -1).sum(1)
    return reg
```
I tried to pass value to these two kinds of implementation and get the same answer.

**Comment by Johnson-yue** [comment]
Oh, I see, and what is your result ?? good or bad?
